osu.Game/Graphics/UserInterface/BackButton.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) 2007-2016 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu-framework/master/LICENCE
+
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Transformations;
+using osu.Framework.Input;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    // Basic back button as it was on stable (kinda). No skinning possible for now
+    public class BackButton : ClickableContainer
+    {
+        private TextAwesome icon;
+
+        private Container leftContainer;
+        private Container rightContainer;
+
+        private const double transform_time = 300.0;
+        private const int pulse_length = 250;
+
+        private const float shear = 0.1f;
+
+        private const int extend_length = 60;
+        private const int initial_extend_length = 40;
+
+        private const int width_extended = 140;
+        private const int width_retracted = 120;
+
+        public BackButton()
+        {
+            RelativeSizeAxes = Axes.None;
+            Width = width_retracted;
+            Height = 50; // same as bottomToolHeight in PlaySongSelect
+
+            Children = new Drawable[]
+            {
+                leftContainer = new Container
+                {
+                    Origin = Anchor.TopLeft,
+                    Anchor = Anchor.TopLeft,
+                    RelativeSizeAxes = Axes.Y,
+                    Width = initial_extend_length,

```

Comment:
how about giving this a relative size and then `ResizingTo()` on the full button instead?